packages/plugin-ext/src/main/browser/workspace-main.ts

```diff
@@ -26,48 +26,53 @@ import { FileStat } from '@theia/filesystem/lib/common';
 import { FileSearchService } from '@theia/file-search/lib/common/file-search-service';
 import URI from '@theia/core/lib/common/uri';
 import { WorkspaceService } from '@theia/workspace/lib/browser';
-import { Resource } from '@theia/core/lib/common/resource';
 import { Disposable, DisposableCollection } from '@theia/core/lib/common/disposable';
-import { Emitter, Event, ResourceResolver } from '@theia/core';
 import { FileWatcherSubscriberOptions } from '../../common/plugin-api-rpc-model';
 import { InPluginFileSystemWatcherManager } from './in-plugin-filesystem-watcher-manager';
 import { PluginServer } from '../../common/plugin-protocol';
 import { FileSystemPreferences } from '@theia/filesystem/lib/browser';
+import { TextContentResourceResolver } from './text-content-resource';
+import { RPCProtocolServiceProvider } from './main-context';
 
+@injectable()
 export class WorkspaceMainImpl implements WorkspaceMain, Disposable {
 
-    private readonly proxy: WorkspaceExt;
+    private proxy: WorkspaceExt;
 
     private storageProxy: StorageExt;
 
+    @inject(MonacoQuickOpenService)
     private quickOpenService: MonacoQuickOpenService;
 
+    @inject(FileSearchService)
     private fileSearchService: FileSearchService;
 
+    @inject(InPluginFileSystemWatcherManager)
     private inPluginFileSystemWatcherManager: InPluginFileSystemWatcherManager;
 
     private roots: FileStat[];
 
+    @inject(TextContentResourceResolver)
     private resourceResolver: TextContentResourceResolver;
 
+    @inject(PluginServer)
     private pluginServer: PluginServer;
 
+    @inject(WorkspaceService)
     private workspaceService: WorkspaceService;
 
+    @inject(FileSystemPreferences)
     private fsPreferences: FileSystemPreferences;
 
+    @inject(RPCProtocol)
+    private readonly rpc: RPCProtocol;
+
     protected readonly toDispose = new DisposableCollection();
 
-    constructor(rpc: RPCProtocol, container: interfaces.Container) {
-        this.proxy = rpc.getProxy(MAIN_RPC_CONTEXT.WORKSPACE_EXT);
-        this.storageProxy = rpc.getProxy(MAIN_RPC_CONTEXT.STORAGE_EXT);
-        this.quickOpenService = container.get(MonacoQuickOpenService);
-        this.fileSearchService = container.get(FileSearchService);
-        this.resourceResolver = container.get(TextContentResourceResolver);
-        this.pluginServer = container.get(PluginServer);
-        this.workspaceService = container.get(WorkspaceService);
-        this.fsPreferences = container.get(FileSystemPreferences);
-        this.inPluginFileSystemWatcherManager = container.get(InPluginFileSystemWatcherManager);
+    @postConstruct()
+    protected init(): void {
+        this.proxy = this.rpc.getProxy(MAIN_RPC_CONTEXT.WORKSPACE_EXT);
```

Comment:
We could use dynamic bindings to inject proxies instead of using `postConsturct`. But could be done with another PR.